stl/inc/vector

```diff
@@ -218,7 +218,7 @@ public:
     }
 
     _CONSTEXPR20 void _Seek_to(const value_type* _It) noexcept {
-        _Ptr = _Refancy<_Tptr>(const_cast<value_type*>(_It));
+        _Ptr = _It == nullptr ? _Tptr() : _Refancy<_Tptr>(const_cast<value_type*>(_It));

```

Comment:
I'd like to avoid this check when `_Tptr` is a pointer type. I think we should either do this check in the fancy overload of `_Refancy`, or introduce a new `_Refancy_maybe_null` (a la `_Unfancy_maybe_null`) overload set so only this callsite performs the null check with only fancy pointers.

Reply:
I've found a case where `basic_string::const_iterator` also needs to be fixed.

It seems the new mechanism is seldomly used (only in 2 `_Seek_to` functions). Should we just expand `_Refancy_maybe_null` into `_Seek_to` instead of adding new functions?

Reply:
I prefer `_Refancy_maybe_null` to repeating the code in two places, even though it's small, so the semantic distinction is clear. Not to mention that we may eventually find more places.